framework/src/ble_manager/ble_manager_lwnl_listener.c

```diff
@@ -92,16 +115,17 @@ int lwnl_fetch_ble_event(int fd, void *buf, int buflen)
 	memcpy(&status, type_buf, sizeof(lwnl_cb_status));
 	memcpy(&len, type_buf + sizeof(lwnl_cb_status), sizeof(uint32_t));
 
-	BLE_LOG_VERBOSE("dev state(%d) length(%d)\n", status.evt, len);
+	BLE_LOG_INFO("dev state(%d) length(%d)\n", status.evt, len);
 	if (status.type == LWNL_DEV_BLE) {
 		_lwnl_call_event(fd, status, len);
 		hmsg->msg = &g_msg;
 	} else {
 		/* Remove wifi data in socket. */
 		char *t_buf = (char *)malloc(len);
-		nbytes = read(fd, (char *)t_buf, len);
-		BLE_LOG_INFO("read unused data nbytes : %d / len : %d\n", nbytes, len);
-		free(t_buf);
+		if (t_buf) {
+			nbytes = read(fd, (char *)t_buf, len);
+			free(t_buf);
```

Comment:
After reading the data into `buf`, free it immediately.
Is this correct?
And if it is correct, why do we need to read?